readthedocs/core/utils/__init__.py

```diff
@@ -153,3 +154,16 @@ def slugify(value, *args, **kwargs):
 
 
 slugify = allow_lazy(slugify, six.text_type, SafeText)
+
+
+def safe_makedirs(directory_name):
+    """
+    Makedirs has an issue where it has a race condition around
+    checking for a directory and then creating it.
+    This catches the exception in this case.
+    """
+
+    try:
+        os.makedirs(directory_name)
+    except OSError:
```

Comment:
Is this strictly catching `makedirs` created dirs that already exist? It seems that `OSError` would be thrown in a number of cases that we might care about.